models/buffer_disk.go

```diff
@@ -205,15 +215,27 @@ func (b *DiskBuffer) resetBatch() {
 }
 
 // This is very messy and not ideal, but serves as the only way I can find currently
-// to actually clear the walfile completely if needed, since Truncate() calls require
+// to actually treat the walfile as empty if needed, since Truncate() calls require
 // that at least one entry remains in them otherwise they return an error.
 // Related issue: https://github.com/tidwall/wal/issues/20
-func (b *DiskBuffer) resetWalFile() {
-	b.file.Close()
-	os.Remove(b.path)
-	walFile, err := wal.Open(b.path, nil)
-	if err != nil {
-		panic(err)
+func (b *DiskBuffer) handleEmptyFile() {
+	if b.isEmpty {
+		err := b.file.TruncateFront(b.readIndex() + 1)
+		if err != nil {
+			log.Printf("E! readIndex: %d, buffer len: %d", b.readIndex(), b.length())
+			panic(err)
+		}
+		b.isEmpty = false
+	}
```

Comment:
This condition should be removed on the caller side and the code should be simplified to
```suggestion
	if b.isEmpty {
		return
	}
	if err := b.file.TruncateFront(b.readIndex() + 1); err != nil {
		log.Printf("E! readIndex: %d, buffer len: %d", b.readIndex(), b.length())
		panic(err)
	}
	b.isEmpty = false
```